x/auctions/types/genesis.go

```diff
@@ -0,0 +1,25 @@
+// Copyright Tharsis Labs Ltd.(Evmos)
+// SPDX-License-Identifier:ENCL-1.0(https://github.com/evmos/evmos/blob/main/LICENSE)
+
+package types
+
+// DefaultGenesisState sets default auctions genesis state.
+func DefaultGenesisState() *GenesisState {
+	return &GenesisState{
+		Params: Params{
+			EnableAuction: true,
+		},
+	}
+}
+
+// NewGenesisState creates a new genesis state.
+func NewGenesisState(params Params) *GenesisState {
+	return &GenesisState{
+		Params: params,
+	}
```

Comment:
I saw this is the approach we have for all modules, but why are we returning a reference?

Reply:
I believe it's for memory efficiency especially if the struct is large, passing the reference around avoids copying the entire struct.